ambry-server/src/test/java/com.github.ambry.server/Http2BlockingChannel.java

```diff
@@ -96,31 +103,39 @@ public void disconnect() throws IOException {
 
   @Override
   public void send(Send request) throws IOException {
+    System.out.println("send");
     ByteBufferChannel byteBufferChannel = new ByteBufferChannel(ByteBuffer.allocate((int) request.sizeInBytes()));
     while (!request.isSendComplete()) {
       request.writeTo(byteBufferChannel);
     }
     byteBufferChannel.getBuffer().position(0);
     ByteBuf byteBuf = Unpooled.wrappedBuffer(byteBufferChannel.getBuffer());
 
-    Http2ClientStreamInitializer initializer = new Http2ClientStreamInitializer(http2ResponseHandler);
-    Http2StreamChannel childChannel =
-        new Http2StreamChannelBootstrap(channel).handler(initializer).open().syncUninterruptibly().getNow();
+    Http2StreamChannel childChannel = http2StreamChannelBootstrap.open().syncUninterruptibly().getNow();
     Http2Headers http2Headers = new DefaultHttp2Headers().method(HttpMethod.POST.asciiName()).scheme("https").path("/");
+    http2Headers.set(RestUtils.Headers.HTTP2_FRONTEND_REQUEST, "true");
+    channelPromise = childChannel.newPromise();
```

Comment:
Instead of combining ambry callbacks and netty promises here, how about setting the attribute to a `CompletableFuture` or ambry's `FutureResult`? Then you can wait on the future without requiring lines 135-137